### DISCUSSION TITLE
Does it work with server-side encryption?

### DISCUSSION THREAD

**Discussion by tio-trom** [nextcloud/recognize#577] [General]
I have enabled server side encryption and I wonder if it can scan any files considering they are encrypted across multiple users.

**Comment by erzacorp** [comment]
Not for me.

**Comment by marcelklehr** [comment]
@juliushaertl What would I need to change to support server-side encryption?

**Reply by juliusknorr** [reply]
Depends a bit if there is a master key enabled (default) or user keys.

User keys will never work as the user password is needed and the files cannot be accessed through the background job. For master key encryption this should in theory work transparently if the regular OCP files api methods are used. But hard to tell without actually looking into what is happening currently.
